app.js

```diff
@@ -0,0 +1,108 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+var WATCH_DELAY = 300;
+
+var indexStatic;
+var router;
+var app = express();
+
+var startApp = function() {
+  app.listen(PORT, function() {
+    console.log('Listening on port', PORT);
+  });
+};
+
+var startDevApp = function() {
+  var gulp = require('gulp');
+  var gulpfile = require('./gulpfile');
+
+  require('./lib/developer-help')();
+
+  indexStaticWatcher.watch(WATCH_DELAY, function(newIndexStatic) {
+    updateIndexStatic(newIndexStatic);
+    console.log('Rebuilt server-side bundle.');
+  });
+
+  gulp.start('watch-webpack');
```

Comment:
One of the things I had a hard time discovering is that `gulp` is actually a subclass of [`Orchestrator`](https://github.com/orchestrator/orchestrator#readme). So it has methods like `start()` that aren't actually documented in the [Gulp API docs](https://github.com/gulpjs/gulp/blob/master/docs/API.md).